packages/core/phase3/lib/pncUpdateRequestGenerators/normalDisposalGenerator/normalDisposalGenerator.test.ts

```diff
@@ -0,0 +1,240 @@
+          FullCode: "RT88191"
+        }
+      } as unknown as OffenceReason
+    },
+    AddedByTheCourt: true,
+    ActualOffenceStartDate: {
+      StartDate: new Date("2025-01-01")
+    },
+    Result: [
+      {
+        NextHearingDate: "2024-12-11T10:11:12.000Z",
+        NextResultSourceOrganisation: {
+          TopLevelCode: "B",
+          SecondLevelCode: "01",
+          ThirdLevelCode: "00",
+          BottomLevelCode: "00",
+          OrganisationUnitCode: "B000000"
+        },
+        PNCDisposalType: 2059,
+        PNCAdjudicationExists: true,
+        ResultClass: ResultClass.ADJOURNMENT_PRE_JUDGEMENT,
+        ResultQualifierVariable: [{ Code: "LE" }]
+      }
+    ] as Result[]
+  } as Offence
+
+  const pncUpdateDataset = generatePncUpdateDatasetFromOffenceList([offence])
+  pncUpdateDataset.AnnotatedHearingOutcome.HearingOutcome.Case.HearingDefendant = {
+    ...pncUpdateDataset.AnnotatedHearingOutcome.HearingOutcome.Case.HearingDefendant,
+    ArrestSummonsNumber: "1101ZD0100000410780J",
+    RemandStatus: "CB",
+    BailConditions: ["This is a dummy bail condition."]
+  }
+  pncUpdateDataset.AnnotatedHearingOutcome.HearingOutcome.Case = {
+    ...pncUpdateDataset.AnnotatedHearingOutcome.HearingOutcome.Case,
+    ForceOwner: {
+      TopLevelCode: "A",
+      SecondLevelCode: "02",
+      ThirdLevelCode: "BJ",
+      BottomLevelCode: "01",
+      OrganisationUnitCode: "A02BJ01"
+    },
+    PTIURN: "01ZD0303208",
+    CourtCaseReferenceNumber: "97/1626/008395Q"
+  }
+  pncUpdateDataset.AnnotatedHearingOutcome.HearingOutcome.Hearing = {
+    ...pncUpdateDataset.AnnotatedHearingOutcome.HearingOutcome.Hearing,
+    DateOfHearing: new Date("2024-12-05"),
+    CourtHearingLocation: {
+      TopLevelCode: "B",
+      SecondLevelCode: "01",
+      ThirdLevelCode: "00",
+      BottomLevelCode: "00",
+      OrganisationUnitCode: "B000000"
+    },
+    CourtType: "MCA",
+    CourtHouseName: "Magistrates' Courts London Croydon"
+  }
+
+  return pncUpdateDataset
+}
+
+const operation = {
+  code: "DISARR",
+  data: {
+    courtCaseReference: "97/1626/008395Q"
+  }
+} as Operation<PncOperation.NORMAL_DISPOSAL>
+
+describe("normalDisposalGenerator", () => {
+  beforeEach(() => {
+    mockedLookupOrganisationUnitByCode.mockRestore()
+  })
+
+  it("should return error when ASN is invalid", () => {
+    const pncUpdateDataset = createPncUpdateDataset()
+    pncUpdateDataset.AnnotatedHearingOutcome.HearingOutcome.Case.HearingDefendant.ArrestSummonsNumber =
+      "1101ZD0100000410780JXX"
+
+    const result = normalDisposalGenerator(pncUpdateDataset, operation)
+
+    expect(isError(result)).toBe(true)
+    expect((result as Error).message).toBe("Invalid ASN length. Length is 22")
+  })
+
+  it("should return error when court case reference in operation is invalid", () => {
+    const pncUpdateDataset = createPncUpdateDataset()
+    const operationWithInvalidCourtCaseReference = {
+      ...structuredClone(operation),
```

Comment:
Do we have to use `structuredClone` if we're spreading into a new object?

Reply:
The reason is spreading (shallow clone) doesn't clone nested objects. In this case, the `Result` array references the original `Result` array. But with `structureClone` we deep clone the object.

Reply:
But I think operation type is simple and we can just create new object without cloning here:
```
    const operationWithInvalidCourtCaseReference = {
      code: "DISARR",
      data: {
        courtCaseReference: "97/1626/008395"
      }
    }

```

Reply:
Ahhhh okay, that makes sense - good spot on simplifying it!